cfme/tests/middleware/test_middleware_provider.py

```diff
@@ -25,56 +25,62 @@
 @pytest.mark.usefixtures('has_no_middleware_providers')
 def test_provider_add_with_bad_credentials(provider):
     """ Tests provider add with bad credentials"""
-    provider.credentials['default'] = Credential(
+    provider.endpoints['default'].credentials = Credential(
         principal='bad',
         secret='reallybad'
     )
     with error.expected('Credential validation was not successful: Invalid credentials'):
         provider.create(validate_credentials=True)
-    assert (not provider.add_provider_button.can_be_clicked)
+    view = provider.create_view(MiddlewareProviderAddView)
+    assert not view.add.active
 
 
 def test_add_cancelled_validation():
     """Tests that the flash message is correct when add is cancelled."""
     prov = HawkularProvider()
     prov.create(cancel=True)
-    flash.assert_message_match('Add of Middleware Provider was cancelled by the user')
+    view = prov.browser.create_view(MiddlewareProviderAddView)
+    view.flash.assert_success_message('Add of Middleware Provider was cancelled by the user')
 
 
-def test_password_mismatch_validation(soft_assert):
+def test_password_mismatch_validation(provider, soft_assert):
     """ Tests password mismatch check """
-    prov = HawkularProvider()
-    cred = Credential(
+    provider.endpoints['default'].credentials = Credential(
         principal='bad',
         secret=fauxfactory.gen_alphanumeric(5),
         verify_secret=fauxfactory.gen_alphanumeric(6)
     )
-    navigate_to(prov, 'Add')
-    fill(prov.properties_form, {"type_select": "Hawkular", "hostname_text": "test",
-                                "name_text": "test", "port_text": "8080"})
-    fill(cred.form, cred)
-    sel.wait_for_ajax()
-    soft_assert(not form_buttons.validate.can_be_clicked)
-    soft_assert(not prov.add_provider_button.can_be_clicked)
-    soft_assert(cred.form.default_verify_secret.angular_help_block == "Passwords do not match")
+    add_view = navigate_to(provider, 'Add')
```

Comment:
you already have prepared provider object. You can copy this object, override default endpoint credentials and call its create() method. You won't need to play with all these fills then